bin/build-dist

```diff
@@ -2,32 +2,86 @@
 
 set -eux
 
+PREFER_AUTHORING=${PREFER_AUTHORING:='1'}
 BUILD_DIR=${BUILD_DIR:='build_dir'}
+INSTALL_DIR=
+
+case "$1" in
+	--favour-authoring | --prefer-authoring | --authoring)
+		PREFER_AUTHORING="1"
+		;;
+	--disfavour-authoring | --prefer-builder | --builder )
+		PREFER_AUTHORING="0"
+		;;
+esac
+
+function error () {
+	echo >&2 "$@"
+	return 1
+}
+
+function last-modified-dir () {
+    ls -td -- */ | head -n 1 | cut -d'/' -f1
+}
+
+function module-build-install-dir () {
+	perl -MModule::Build -e'print Module::Build->resume->dist_dir;'
+}
+
+function authoring-tool-favoured () {
+	case "$FAVOUR_AUTHORING" in
+		0 | false) return 0 ;;
+		1 | true ) return 1 ;;
+		*)
+			error "Unknown value of FAVOUR_AUTHORING - '$FAVOUR_AUTHORING'"
+			;;
+	esac
+}
+
+function build-with-dist-zilla () {
+	[ -e dist.ini ] || return 1
```

Comment:
Please always use the `[[ .. ]]` form of testing.
It is Bash syntax and much more reliable than `[` which is essentially a function call and requires all the rigidity of calling functions right in Bash.
For instance `[[ $foo ]]` needs no quotes around `$foo` because its a Bash expression and Bash knows how to expand vars as intended there.